Tests/backend-specific-tests/Synapse/RunnerSynapseTest.php

```diff
@@ -0,0 +1,154 @@
+use Keboola\Temp\Temp;
+
+class RunnerSynapseTest extends BaseRunnerTest
+{
+    private function clearBuckets()
+    {
+        foreach (['in.c-synapse-runner-test', 'out.c-synapse-runner-test'] as $bucket) {
+            try {
+                $this->getClient()->dropBucket($bucket, ['force' => true]);
+            } catch (ClientException $e) {
+                if ($e->getCode() != 404) {
+                    throw $e;
+                }
+            }
+        }
+    }
+
+    protected function initStorageClient()
+    {
+        $this->client = new Client([
+            'url' => STORAGE_API_URL_SYNAPSE,
+            'token' => STORAGE_API_TOKEN_SYNAPSE,
+        ]);
+    }
+
+    public function setUp()
+    {
+        if (!RUN_SYNAPSE_TESTS) {
+            return;
+        }
+        parent::setUp();
+    }
+
+    private function createBuckets()
+    {
+        $this->clearBuckets();
+        // Create buckets
+        $this->getClient()->createBucket('synapse-runner-test', Client::STAGE_IN, 'Docker TestSuite', 'synapse');
+        $this->getClient()->createBucket('synapse-runner-test', Client::STAGE_OUT, 'Docker TestSuite', 'synapse');
+    }
+
+    /**
+     * @param array $componentData
+     * @param $configId
+     * @param array $configData
+     * @param array $state
+     * @return JobDefinition[]
+     */
+    protected function prepareJobDefinitions(array $componentData, $configId, array $configData, array $state)
+    {
+        $jobDefinition = new JobDefinition($configData, new Component($componentData), $configId, 'v123', $state);
+        return [$jobDefinition];
+    }
+
+    public function testWorkspaceMapping()
+    {
+        if (!RUN_SYNAPSE_TESTS) {
+            self::markTestSkipped('Synapse test is disabled.');
+        }
+        $this->clearBuckets();
+        $this->createBuckets();
+        $temp = new Temp();
+        $temp->initRunFolder();
+        $csv = new CsvFile($temp->getTmpFolder() . '/upload.csv');
+        $csv->writeRow(['id', 'text']);
+        $csv->writeRow(['test1', 'test1']);
+        $this->getClient()->createTableAsync('in.c-synapse-runner-test', 'mytable', $csv);
+        unset($csv);
+
+        $componentData = [
+            'id' => 'keboola.runner-workspace-synapse-test',
+            'data' => [
+                'definition' => [
+                    'type' => 'aws-ecr',
+                    'uri' => '061240556736.dkr.ecr.us-east-1.amazonaws.com/keboola.runner-workspace-test',
+                    'tag' => 'latest',
+                ],
+                'staging_storage' => [
+                    'input' => 'workspace-synapse',
+                    'output' => 'workspace-synapse',
+                ],
+            ],
+            // https://keboola.slack.com/archives/C02C3GZUS/p1598942156005100
+            // https://github.com/microsoft/msphpsql/issues/400#issuecomment-481722255
+            'features' => ['container-root-user'],
```

Comment:
Takze tymto sa to pusti pod rootom a to riesi ten UID problem?

Reply:
jj

Reply:
pravdepodobne to bude problem i u transformaci, ale to si vyzkousim az v produkci a bud se to bude muset fixnout v te komponente nebo poustet pod rootem